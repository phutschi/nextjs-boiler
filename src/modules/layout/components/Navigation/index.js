/* global React */

import Link from 'next/link';
import { withRouter } from 'next/router';

import classNames from 'classnames';

import routes from '../../../../routes';

import theme from './main.scss';

const Navigation = ( { router } ) => {
    const handleLinkClick = href => event => {
        event.preventDefault();
        router.push( href );
    };

    const navigationWrapperClassnames = classNames( theme.navigation );

    return (
        <div className={navigationWrapperClassnames}>
            {
                Object.keys( routes ).map( route => {
                    const linkClassnames = classNames( theme.link, { [ theme.active ] : router.pathname === routes[ route ] } );
                    const path = routes[ route ];

                    return (
                        <Link key={path} href={path}>
                            <a
                                href="#"
                                onClick={handleLinkClick( path )}
                                className={linkClassnames}
                            >
                                {route.charAt( 0 ).toUpperCase() + route.slice( 1 )}
                            </a>
                        </Link>
                    );
                } )
            }
        </div>
    );
};

export default withRouter( Navigation );
