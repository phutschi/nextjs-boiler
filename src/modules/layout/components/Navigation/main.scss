.navigation {
  text-align: center;
  position: absolute;
  bottom: 20px;
  left: 0;
  right: 0;

  .link {
    color: #000;
    cursor: pointer;
    font-size: 12px;
    margin: 0 6px;
    padding: 3px 2px 0;
    text-decoration: none;
    background-image: linear-gradient(120deg, #cfd9df 0%, #e2ebf0 100%);
    background-repeat: no-repeat;
    background-size: 100% .1em;
    background-position: 0 110%;
    transition: background-size 0.15s ease-in, color .3s ease;

    &:hover {
      background-size: 100% 88%;
    }

    &.active {
      background-size: 100% .1em;
      background-position: 0 100%;
      pointer-events: none;
      cursor: default;
    }
  }
}
