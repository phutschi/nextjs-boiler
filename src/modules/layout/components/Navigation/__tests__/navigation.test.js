/* global expect */

import * as React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Navigation from '../';

describe( '<Navigation />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <Navigation /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );
} );
