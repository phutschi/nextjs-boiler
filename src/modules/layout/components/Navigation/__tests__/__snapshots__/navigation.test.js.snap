// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snapshot 1`] = `
<div
  className="navigation"
>
  <a
    className="link"
    href="#"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="link"
    href="#"
    onClick={[Function]}
  >
    Posts
  </a>
  <a
    className="link"
    href="#"
    onClick={[Function]}
  >
    Counter
  </a>
</div>
`;
